syslog-ng-rs/syslog-ng-common/src/proxies/parser/proxy.rs

```diff
@@ -29,6 +29,15 @@ impl<B> ParserProxy<B> where B: ParserBuilder<LogParser>
         }
     }
 
+    pub fn with_parser_and_builder(builder: Option<B>,
+                                   parser: Option<B::Parser>)
+                                   -> ParserProxy<B> {
+        ParserProxy {
+            parser: parser,
```

Comment:
That's right :) I'd like to stick to Rust idioms in Rust projects. In C, this would be bad.